framework/web/User.php

```diff
@@ -562,6 +544,49 @@ protected function sendIdentityCookie($identity, $duration)
     }
 
     /**
+     * Determines if an identity cookie format is valid and has a valid auth key.
+     * This method is used when [[enableAutoLogin]] is true.
+     * This method attempts to authenticate a user using the information in the identity cookie.
+     * @return array|null Returns an array of 'identity' and 'duration' if valid, otherwise null.
+     * @see loginByCookie()
+     */
+    protected function validateIdentityCookie()
+    {
+        $value = Yii::$app->getRequest()->getCookies()->getValue($this->identityCookie['name']);
+        if ($value === null) {
+            return null;
+        }
+        $data = json_decode($value, true);
+        if (count($data) === 3 && isset($data[0], $data[1], $data[2])) {
+            list ($id, $authKey, $duration) = $data;
+            /* @var $class IdentityInterface */
+            $class = $this->identityClass;
+            $identity = $class::findIdentity($id);
+            if ($identity !== null) {
+                if (!$identity instanceof IdentityInterface) {
+                    throw new InvalidValueException("$class::findIdentity() must return an object implementing IdentityInterface.");
+                } elseif (!$identity->validateAuthKey($authKey)) {
+                    Yii::warning("Invalid auth key attempted for user '$id': $authKey", __METHOD__);
+                } else {
+                    return [ 'identity' => $identity, 'duration' => $duration ];
+                }
+            }
+        }
+        $this->removeIdentityCookie();
+        return null;
+    }
+     
+    /**
+     * Removes the identity cookie.
+     * This method is used when [[enableAutoLogin]] is true.
+     * @param IdentityInterface $identity
```

Comment:
Thanks!  I had originally intended to pass $identity in case it was needed, but someone told me that was a bad idea.  So I took it out and neglected to update the comment.